x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/script/CartesianPointFieldScript.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.spatial.script;
+
+import org.apache.lucene.geo.XYEncodingUtils;
+import org.apache.lucene.index.LeafReaderContext;
+import org.elasticsearch.script.AbstractPointFieldScript;
+import org.elasticsearch.script.CompositeFieldScript;
+import org.elasticsearch.script.ScriptContext;
+import org.elasticsearch.script.ScriptFactory;
+import org.elasticsearch.search.lookup.SearchLookup;
+import org.elasticsearch.xpack.spatial.common.CartesianPoint;
+
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/**
+ * Script producing cartesian points.
+ */
+public abstract class CartesianPointFieldScript extends AbstractPointFieldScript<CartesianPoint> {
+    public static final ScriptContext<Factory> CONTEXT = newContext("point_field", Factory.class);
+
```

Comment:
I see this is not used anywhere, maybe we should remove it until we decide to have runtime support

Reply:
Agreed. I also remember we discussed with the painless team the option of building a completely different fields API for script for this, so none of the prototype code here is of value. I created an issue to discuss that at https://github.com/elastic/elasticsearch/issues/90182

Reply:
Fixed in https://github.com/elastic/elasticsearch/pull/89216/commits/3fa56032243e566da4f8f4c70beb7f964531a1ad